internal/cmd/cmd.go

```diff
@@ -101,11 +103,25 @@ func defaultAPIClient() (*api.Client, error) {
 		server = envServer
 	}
 
-	return apiClient(server, accessKey, httpTransportForHostConfig(config)), nil
+	ignoreVersion := false
+	if envIgnoreVersion, ok := os.LookupEnv("INFRA_SKIP_VERSION_CHECK"); ok {
+		ignoreVersion, err = strconv.ParseBool(envIgnoreVersion)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse INFRA_SKIP_VERSION_CHECK from environment as a boolean: %w", err)
+		}
+	}
+	if flags != nil {
+		ignoreVersion, err = flags.GetBool("skip-version-check")
+		if err != nil {
+			return nil, fmt.Errorf("failed to read cmd flags for client: %w", err)
+		}
+	}
+
+	return apiClient(server, accessKey, httpTransportForHostConfig(config), ignoreVersion)
 }
 
-func apiClient(host string, accessKey string, transport *http.Transport) *api.Client {
-	return &api.Client{
+func apiClient(host string, accessKey string, transport *http.Transport, ignoreVersion bool) (*api.Client, error) {
```

Comment:
Perhaps to help, we can merge this and above into a single constructor that takes an options struct:

```
func NewAPIClient(&APIClientOptions{
  Host: "...",
  accessKey: "...",
  FromConfig: true,
  IgnoreVersion: true
})
```

Reply:
Even better, this would be simple & flat, without any constructors:

```
// uses default config
client := api.Client{}
```

```
// ignores version
client := api.Client{IgnoreVersion: true}
```

```
// custom for login
client := api.Client{URL}
```